src/main/java/persistence/core/EntityMetadata.java

```diff
@@ -0,0 +1,94 @@
+package persistence.core;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.Table;
+import persistence.exception.PersistenceException;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class EntityMetadata<T> {
+
+    private final String tableName;
+    private final EntityColumns columns;
+    private final EntityColumn idColumn;
+
+    public EntityMetadata(final Class<T> clazz) {
+        this.validate(clazz);
+        this.tableName = initTableName(clazz);
+        this.columns = new EntityColumns(clazz);
+        this.idColumn = this.columns.getId();
+    }
+
+    private void validate(final Class<T> clazz) {
+        if (!clazz.isAnnotationPresent(Entity.class)) {
+            throw new PersistenceException(clazz.getName() + "은 Entity 클래스가 아닙니다.");
+        }
+    }
+
+    private String initTableName(final Class<?> clazz) {
+        final Table tableAnnotation = clazz.getDeclaredAnnotation(Table.class);
+        return Optional.ofNullable(tableAnnotation)
+                .filter(table -> !table.name().isEmpty())
+                .map(Table::name)
+                .orElseGet(clazz::getSimpleName);
+    }
+
+    public String getTableName() {
+        return this.tableName;
+    }
+
+    public EntityColumns getColumns() {
+        return this.columns;
+    }
+
+    public List<String> getInsertableColumnNames() {
+        return this.columns.stream()
+                .filter(EntityColumn::isInsertable)
+                .map(EntityColumn::getName)
+                .collect(Collectors.toUnmodifiableList());
+    }
+
+    public List<String> getInsertableColumnFieldNames() {
+        return this.columns.stream()
+                .filter(EntityColumn::isInsertable)
+                .map(EntityColumn::getFieldName)
+                .collect(Collectors.toUnmodifiableList());
+    }
+
+    public String getIdColumnName() {
+        return this.idColumn.getName();
+    }
+
+    public String getIdColumnFieldName() {
+        return this.idColumn.getFieldName();
+    }
+
+    public int getColumnSize() {
+        return this.columns.size();
+    }
+
+    public List<String> getColumnNames() {
+        return this.columns.stream()
+                .map(EntityColumn::getName)
+                .collect(Collectors.toUnmodifiableList());
+    }
+
+    public List<String> getColumnFieldNames() {
+        return this.columns.stream()
+                .map(EntityColumn::getFieldName)
+                .collect(Collectors.toUnmodifiableList());
+    }
+
+    public EntityColumn getIdColumn() {
+        return this.idColumn;
+    }
+
+    public EntityColumns getInsertableColumn() {
```

Comment:
```suggestion
    public EntityColumns toInsertableColumns() {
```
요런 느낌은 어떤가요? 

영어의 전치사 참 어려운 부분인것 같아요... ㅋㅋ 
하지만, 잘만 활용한다면 의미를 다채롭게 표현할수 있지 않을까요? 🤔 
[method naming 에서 of 와 from 의 차이는 뭘까?](https://wooobo.github.io/post/2021-11-09-method-naming-conventions/)

Reply:
저도 헷갈리던 of 와 from 의 차이를 알게되었네요!

사실 이 부분은 1단계 미션 진행시 고민했던 부분인데요. 불변객체 일급컬렉션이다보니 메서드 호출시 항상 같은 객체를 리턴하게 되는데 그런면에서 필드로 그냥 먼저 가지고 있다가 반환할지말지 고민이었답니다. 만약 필드로 미리 가지고있는 값이라면 get 이르는 네이밍은 괜찮을까요?

Reply:
사실 `get-`이라고 해도 상관없어요! ㅎ
`get-`을 붙인다면 정말 단순히 내부 값을 외부로 보내지는 형태로만 사용할것 같아요.
```
    public String getTableName() {
        return this.tableName;
    }
```
이정도느낌?으로요 😄 

`public EntityColumns getInsertableColumn() {`
내부의 값으로 새로운 객체를 만드는것이므로,
`buildInsertableColumns`, `toInsertableColumns` 이런 느낌의 네이밍이 어울릴것 같아요.

저의 개인적인 관점이므로 정답은 없습니다 ㅎㅎ

Reply:
실무에서 `get-`을 안쓰기 어렵죠 ㅜㅜ
`get`을 안쓰고 개발하라니... 정말 가혹하다고 생각합니다.

`왜 `get`으로 값을 꺼내올려고하지?` 한번씩 생각해주시면서 진행해보시는건 어떨까요?
객체를 단순한 자료구조가 아닌 정말 책임과 역할이 있는 객체로 대우를 해주는것이죠!